tests/timeseries/test_base_types.py

```diff
@@ -0,0 +1,79 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for battery pool."""
+
+
+from datetime import datetime
+
+from frequenz.sdk.timeseries._base_types import Bounds, SystemBounds
+from frequenz.sdk.timeseries._quantities import Power
+
+
+def test_bounds_contains() -> None:
+    """
+    Test the `__contains__` method of the `Bounds` class.
+
+    This method checks if a value is within the defined bounds.
+    """
+    bounds = Bounds(lower=Power.from_watts(10), upper=Power.from_watts(100))
+
+    assert Power.from_watts(50) in bounds  # within bounds
+    assert Power.from_watts(10) in bounds  # at lower bound
+    assert Power.from_watts(100) in bounds  # at upper bound
+    assert Power.from_watts(9) not in bounds  # below lower bound
+    assert Power.from_watts(101) not in bounds  # above upper bound
+
+    bounds_no_lower = Bounds(lower=None, upper=Power.from_watts(100))
+    assert Power.from_watts(50) in bounds_no_lower  # within upper bound
+    assert Power.from_watts(100) in bounds_no_lower  # at upper bound
+    assert Power.from_watts(101) not in bounds_no_lower  # above upper bound
+
+    bounds_no_upper = Bounds(lower=Power.from_watts(10), upper=None)
+    assert Power.from_watts(50) in bounds_no_upper  # within lower bound
+    assert Power.from_watts(10) in bounds_no_upper  # at lower bound
+    assert Power.from_watts(9) not in bounds_no_upper  # below lower bound
+
+    # test succeeds, mypy 'Unsupported operand types for in ("Power" and "Bounds[None]")'
+    # bounds_none = Bounds(lower=None, upper=None)
+    # assert Power.from_watts(15) not in bounds_none
+
```

Comment:
I reopened this because the suggested changes were not made. Even if the extra validation is not added, at least having tests for this would be good:

> I would also add some tests for corner cases, like using plain ints, using an empty range, like Bounds(lower=0, upper=0)

If you think this should not be done, it would be good to give a reason.